### DISCUSSION TITLE
defaultStorageProvider=?

### DISCUSSION THREAD

**Discussion by Vladimir-Va** [casdoor/casdoor#1367] [Q&A] [answered]
Hello, casdoor team.
Please clarify what should I specify as a value in the app.conf for docker setup in line
`defaultStorageProvider=`
If a
`defaultStorageProvider=Local`
Then in [Documentation](https://casdoor.org/ru/docs/provider/storage/overview#local) With Local type, the only item that you need to configure is Domain field.
and add new line in app.conf
`Domain=http://127.0.0.1:7001/images`
or i just need setup
`defaultStorageProvider=http://127.0.0.1:7001/images`
and if i setup casdoor as docker setup on `http://127.0.0.1:7001`
casdoor create `/images` folder in container and Resources uploaded to this folder?

**Comment by forestmgy** [comment]
In my opinion, you can set the `defaultStorageProvider` as `http://127.0.0.1:7001/images` 
Because if you use docker to run the Go language program, the folder created in the program will be placed in the path of the container. Specifically, if the folder created in the program is located in the current directory, the location in the container will also be in the current directory. For example, if the /var/tmp/test directory is created in the program, the location of the directory in the container will also be /var/tmp/test.

**Comment by idc77** [comment]
This is so frustrating. Discord, no response. On the web no one had the same issue.
Code is a mess.
What do I have to put there in `defaultStorageProvider`?

**Comment by hsluoyz** [comment, marked as answer]
defaultStorageProvider is the provider name (without owner). 

For example, for this provider: https://door.casdoor.com/providers/admin/provider_storage_local_file_system , it should be:

`defaultStorageProvider = provider_storage_local_file_system`
